packages/build/tests/blobs_upload/tests.js

```diff
@@ -0,0 +1,99 @@
+import { version as nodeVersion } from 'node:process'
+
+import { BlobsServer, getDeployStore } from '@netlify/blobs'
+import { Fixture, normalizeOutput } from '@netlify/testing'
+import test from 'ava'
+import getPort from 'get-port'
+import semver from 'semver'
+import tmp from 'tmp-promise'
+
+const TOKEN = 'test'
+
+test.beforeEach(async (t) => {
+  const port = await getPort()
+  t.context.blobRequestCount = { set: 0, get: 0 }
+
+  const tmpDir = await tmp.dir()
+  t.context.blobServer = new BlobsServer({
+    port,
+    token: TOKEN,
+    directory: tmpDir.path,
+    onRequest: ({ type }) => {
+      t.context.blobRequestCount[type] = (t.context.blobRequestCount[type] || 0) + 1
+    },
+  })
+
+  await t.context.blobServer.start()
+
+  process.env.NETLIFY_BLOBS_CONTEXT = Buffer.from(
+    JSON.stringify({
+      edgeURL: `http://localhost:${port}`,
+    }),
+  ).toString('base64')
+})
+
+test.afterEach.always(async (t) => {
+  await t.context.blobServer.stop()
+  delete process.env.NETLIFY_BLOBS_CONTEXT
+})
+
+test("blobs upload, don't run when deploy id is provided and no files in directory", async (t) => {
+  const output = await new Fixture('./fixtures/src_empty')
+    // Passing `offline: true` to avoid fetching the configuration from the API
+    .withFlags({ deployId: 'abc123', token: TOKEN, offline: true })
+    .runWithBuild()
+
+  t.is(t.context.blobRequestCount.set, 0)
+
+  t.snapshot(normalizeOutput(output))
+})
+
+test("blobs upload, don't run when there are files but deploy id is not provided", async (t) => {
+  const output = await new Fixture('./fixtures/src_with_blobs')
+    .withFlags({ token: TOKEN, offline: true })
+    .runWithBuild()
+
+  t.is(t.context.blobRequestCount.set, 0)
+
+  t.snapshot(normalizeOutput(output))
+})
+
+test.serial('blobs upload, uploads files to deploy store', async (t) => {
```

Comment:
is there a reason why those have to be serial in comparison to the other two on the top?

I think as you get a unique port for each test all can be concurrently. (the default)

Reply:
I was running into flaky tests when not running it serially. But it might have been something else. I'll change and we can always add it back if needed.